odp/core/snowflake.py

```diff
@@ -42,6 +41,15 @@ def get_snowflake_queries(credentials: SnowflakeCredentials, since_days: int) ->
         warehouse=credentials.snowflake_warehouse,
     )
 
+
+def get_snowflake_queries(
+    credentials: SnowflakeCredentials, since_days: int, conn: SnowflakeConnection | None = None
+) -> list[QueryRow]:
```

Comment:
It's not totally clear to me from this method signature what's the link between credentials and conn, or that passing `conn` will cause the credentials to be ignored. Would it be clearer if `conn` was the only argument? What would be the downside to doing that?